Death knight Fixes

syrifgit opened the pull request:
## Frost DK
- Frostbane override was unknown, spellID
  - Fixes https://github.com/Hekili/hekili/issues/5031

- Fixed frostcythe prio variable bad mathing
  - Fixes https://github.com/Hekili/hekili/issues/5033
- Cindy Breath CD
  - Fixes https://github.com/Hekili/hekili/issues/5036

## Unholy DK
### Diseases
Blizzard sneaky change the aura IDs for the other 2 plagues this patch.
- Blood Plague: Remains `55078` for Blood, is now `1235372` for Unholy
- Frost Fever: Remains `55095` for Frost, is now `1235371` for Unholy
Fixes https://github.com/Hekili/hekili/issues/5029
### SimulationCraft
Changed the name of an aura, and also decided on a name for the tier aura.
- `festering_scythe_stack` -> `festering_scythe_stacks`

syrifgit committed:
fix Frost Strike being unknown while frostbane is up

Might need to examine spell overrides more broadly, maybe?

Fixes https://github.com/Hekili/hekili/issues/5031

syrifgit committed:
Unholy had some spellID changes

syrifgit committed:
SimC Syntax

syrifgit committed:
oops

syrifgit committed:
Update DeathKnightFrost.lua

syrifgit committed:
Update DeathKnightUnholy.lua

Hekili commented:
For Frost Strike, I'm seeing a different ID than you had when Frostbane is up -- I don't know if that's because there's another talent interaction.  Where did you get 1228436 for the ID?

Setting `known` to the ID of Frost Strike uses that ID for known checks (don't need the known function you'd written).

```Lua
    -- Talent: Chill your $?$owb==0[weapon with icy power and quickly strike the enemy, dealing $<2hDamage> Frost damage.][weapons with icy power and quickly strike the enemy with both, dealing a total of $<dualWieldDamage> Frost damage.]
    frost_strike = {
        id = function() return buff.frostbane.up and 1228433 or 49143 end,
        texture = function() return buff.frostbane.up and 1273742 or 237520 end,
        known = 49143,
        cast = 0,
        cooldown = 0,
        gcd = "spell",

        spend = 35,
        spendType = "runic_power",
        school = function() if talent.dark_talons.enabled and buff.icy_talons.up then return "shadowfrost" end return "frost" end,

        talent = "frost_strike",
        startsCombat = true,

        cycle = function ()
            if debuff.mark_of_fyralath.up then return "mark_of_fyralath" end
            if death_knight.runeforge.razorice and talent.shattering_blade.disabled and debuff.razorice.at_max_stacks then return "razorice" end
        end,

        handler = function ()
            if talent.obliteration.enabled and buff.pillar_of_frost.up then addStack( "killing_machine" ) end
            if talent.shattering_blade.enabled and debuff.razorice.at_max_stacks then removeDebuff( "target", "razorice" ) end
            -- if debuff.razorice.stack > 5 then applyDebuff( "target", "razorice", nil, debuff.razorice.stack - 5 ) end
            if talent.icy_onslaught.enabled then addStack( "icy_onslaught" ) end
            removeBuff( "frostbane" )
            if death_knight.runeforge.razorice then applyDebuff( "target", "razorice", nil, min( 5, buff.razorice.stack + 1 ) ) end
            if talent.frostreaper.enabled then removeBuff( "frost_reaper" ) end
            -- Legacy / PvP
            if pvptalent.bitter_chill.enabled and debuff.chains_of_ice.up then
                applyDebuff( "target", "chains_of_ice" )
            end

            if conduit.eradicating_blow.enabled then removeBuff( "eradicating_blow" ) end
        end,

        auras = {
            unleashed_frenzy = {
                id = 338501,
                duration = 6,
                max_stack = 5,
            }
        },

        copy = { 1228433, 49143 }
    },
    ```

syrifgit commented:
Uh .. my combatlogs that told me thats the spellID I cast?

<img width="827" height="452" alt="image" src="https://github.com/user-attachments/assets/51c59703-c1db-46c2-9fd2-143dfb57e1d9" />

That's super weird. On the actionbar, it's clearly the ID you posted.

Looks like the whole "strikes twice" is actually 2 entirely separate combatlog cast entries when you click the button. One with the ID you posted, a short delay, then one with the ID I used.

<img width="422" height="552" alt="image" src="https://github.com/user-attachments/assets/32e033d4-6e93-4bd3-aeac-e1aeb91a2da1" />


It also seems to apply 2 different frostbane auras at the same time? The second of which immediately fades

<img width="456" height="55" alt="image" src="https://github.com/user-attachments/assets/9308f4e8-8cbc-4487-92d8-df74f1fd33e0" />

Anyways .. will make the changes. Stupid overengineered spell.

syrifgit committed:
spellID

syrifgit committed:
variable math, breath CD

syrifgit commented:
Ready, updated to include:

- Fixed frostbane ID
- Fixed frostcythe prio variable bad mathing: Fixes https://github.com/Hekili/hekili/issues/5033
- Cindy Breath CD: Fixes https://github.com/Hekili/hekili/issues/5036

Hekili commented:
> Uh .. my combatlogs that told me thats the spellID I cast?

![Untitled](https://github.com/user-attachments/assets/db5c3bff-63e0-4896-87e8-58cc3e1b4b6c)

What we need here is the spell on the action button (which is the spell override):

<img width="263" height="213" alt="image" src="https://github.com/user-attachments/assets/18a205fb-2336-4e65-a597-d27bfb0a4c81" />
